compiler/rustc_type_ir/src/inherent.rs

```diff
@@ -50,7 +51,12 @@ pub trait GenericsOf<I: Interner<GenericsOf = Self>> {
 }
 
 pub trait GenericArgs<I: Interner<GenericArgs = Self>>:
-    Copy + DebugWithInfcx<I> + Hash + Eq + IntoIterator<Item = I::GenericArg>
+    Copy
+    + DebugWithInfcx<I>
+    + Hash
+    + Eq
+    + IntoIterator<Item = I::GenericArg>
+    + Deref<Target: Deref<Target = [I::GenericArg]>>
```

Comment:
:/ `Deref<Target: Deref<Target = [I::GenericArg]>>` is not ideal 🤔 don't have any immediate better ideas though

Reply:
Yes it extremely sucks and also is likely not going to work for r-a, but we should fix it later (either adapt the callsites that rely on this to do something else, or something idk).